converter/src/main/java/org/kaaproject/avro/ui/converter/FormAvroConverter.java

```diff
@@ -232,7 +232,15 @@ private static FormField createFieldFromSchema(RecordField rootRecord, Schema sc
      */
     private static void parseFields(RecordField rootRecord, RecordField formData, Schema schema) throws IOException {
         List<Field> schemaFields = schema.getFields();
+        java.util.Set<String> schemaFieldNames = new java.util.HashSet<>();
+
         for (Field field : schemaFields) {
+            String schemaFieldName = field.name().toLowerCase();
+            if (!schemaFieldNames.contains(schemaFieldName)) {
+                schemaFieldNames.add(schemaFieldName);
+            } else {
+                throw new RuntimeException("Field names are not unique!");
+            }
 
```

Comment:
Change RuntimeException to IllegalArgumentException. Change exception message to indicate which field is duplicate.